hack/ci-e2e-tests.sh

```diff
@@ -0,0 +1,171 @@
+  sudo snap install lxd --channel=$LXD_CHANNEL
+  sudo lxd init --auto
+  sudo usermod --append --groups lxd "$USER"
+}
+
+# Create or ensure the k8s profile exists
+function setup_lxd_profile {
+  lxc profile create k8s || true
+  wget -q $K8S_PROFILE_URL -O $K8S_PROFILE_PATH
+  cat $K8S_PROFILE_PATH | lxc profile edit k8s
+  rm -f $K8S_PROFILE_PATH
+}
+
+# Setup and configure the container
+function setup_container {
+  lxc launch $LXC_IMAGE $CONTAINER_NAME -p default -p k8s
+  until exec_in_container true; do
+    sleep 1
+  done
+
+  exec_in_container "apt update && apt install -y snapd"
+  exec_in_container "systemctl start snapd"
+}
+
+function configure_container_env {
+  if [[ $INFRA_PROVIDER == "aws" ]]; then
+    # Check for clusterawsadm binary
+    exec_in_container "which clusterawsadm" || error_exit "clusterawsadm binary not found in container"
+
+    set +x
+    lxc config set $CONTAINER_NAME environment.AWS_REGION "$AWS_REGION"
+    lxc config set $CONTAINER_NAME environment.AWS_SECRET_ACCESS_KEY "$AWS_SECRET_ACCESS_KEY"
+    lxc config set $CONTAINER_NAME environment.AWS_ACCESS_KEY_ID "$AWS_ACCESS_KEY_ID"
+
+    local aws_creds
+    aws_creds=$(lxc exec "$CONTAINER_NAME" -- bash -c "clusterawsadm bootstrap credentials encode-as-profile")
+
+    lxc config set "$CONTAINER_NAME" environment.AWS_B64ENCODED_CREDENTIALS "$aws_creds"
+    set -x
+  fi
+}
+
+# Main installation and configuration
+function setup_management_cluster {
+  sleep 5
+  exec_in_container "snap install k8s --classic --edge"
+  sleep 1
+  exec_in_container "snap install go --classic"
+  exec_in_container "mkdir -p /root/.kube"
+  exec_in_container "sudo k8s bootstrap"
+  exec_in_container "sudo k8s status --wait-ready"
+  exec_in_container "sudo k8s config > /root/.kube/config"
+}
+
+# Transfer and execute scripts
+function install_tools {
+  tools=(install-clusterctl.sh)
+
+  if [[ $INFRA_PROVIDER == "aws" ]]; then
+    tools+=(install-clusterctlawsadm.sh install-aws-nuke.sh)
+  fi
+
+  for script in "${tools[@]}"; do
+    lxc file push ./"$script" $CONTAINER_NAME/root/"$script"
+    exec_in_container "chmod +x /root/$script && /root/$script"
+  done
+}
+
+function init_clusterctl {
+  configure_container_env # Ensures that the right environment variables are set in the container
+
+  lxc file push ./write-provider-config.sh $CONTAINER_NAME/root/write-provider-config.sh
+  exec_in_container "chmod +x /root/write-provider-config.sh"
+  exec_in_container "mkdir -p /root/.cluster-api"
+  exec_in_container "/root/write-provider-config.sh /root/.cluster-api/clusterctl.yaml $CK8S_PROVIDER_VERSION"
+
+  exec_in_container "clusterctl init -i $INFRA_PROVIDER -b ck8s:$CK8S_PROVIDER_VERSION -c ck8s:$CK8S_PROVIDER_VERSION --config /root/.cluster-api/clusterctl.yaml"
+}
+
+function run_e2e_tests {
+  make USE_EXISTING_CLUSTER=true GINKGO_FOCUS="Workload cluster creation" test-e2e
+}
+
+function cleanup {
+  if [[ $SKIP_CLEANUP == "true" ]]; then
+    return
```

Comment:
maybe we can also print something here.

Reply:
I'm thinking of removing the cleanup function from the script, and having the user run that as a separate step. It reduces the size and complexity of the script a bit.